ntpcheck/checker/runner.go

```diff
@@ -104,6 +104,34 @@ func RunCheck(address string) (*NTPCheckResult, error) {
 	return checker.Run()
 }
 
+// RunNTPData is a simple wrapper to connect to address and run NTPCheck.Run()
+// It gathers extra info about the peers using the unix socket private protocol
+func RunNTPData(address string) (*NTPCheckResult, error) {
```

Comment:
Now it's a carbon copy of `Run()` as far as I can tell.

As the whole split is chrony-specific, maybe we can make this function more specific as well?
Like check the flavour and if it's chrony, then only try connecting over unix socket and maybe call specific `checker.RunNTPData()` (or set special parameter on `ChronyCheck`), and if it's NTPD then just `checker.Run()`.

Reply:
Right, I made it more specific